org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/analysis/SlicingRunnableDevice.java

```diff
@@ -0,0 +1,101 @@
+package org.eclipse.scanning.sequencer.analysis;
+
+import java.util.Arrays;
+
+import org.eclipse.dawnsci.analysis.api.io.IDataHolder;
+import org.eclipse.dawnsci.analysis.api.io.ILoaderService;
+import org.eclipse.january.IMonitor;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.scanning.api.ModelValidationException;
+import org.eclipse.scanning.api.ValidationException;
+import org.eclipse.scanning.api.device.AbstractRunnableDevice;
+import org.eclipse.scanning.api.device.IWritableDetector;
+import org.eclipse.scanning.api.device.models.DeviceRole;
+import org.eclipse.scanning.api.device.models.SlicingModel;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+import org.eclipse.scanning.sequencer.ServiceHolder;
+
+/**
+ * 
+ * This device is one that slices the value of another dataset from the scan file.
+ * This is useful because often inline processing, such as an average may need to be done.
+ * If that is needed it is possible to extend SlicingRunnableDevice to implement the
+ * process method.
+ * 
+ * @author Matthew Gerring
+ * @param <T>
+ *
+ */
+public abstract class SlicingRunnableDevice<T extends SlicingModel> extends AbstractRunnableDevice<T> implements IWritableDetector<T>{
+
+
+
+	public SlicingRunnableDevice() {
+		super(ServiceHolder.getRunnableDeviceService());
+		setLevel(100); // Runs at the end of the cycle by default.
+		setRole(DeviceRole.PROCESSING);
+	}
+	
+	@Override
+	public void validate(T model) throws ValidationException {
+		if (model.getDataFile()==null) throw new ModelValidationException("The data file must be set!", model, "dataFile");
+		if (model.getDetectorName()==null) throw new ModelValidationException("The detector name must be set!", model, "detectorName");

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)